tools/mkspecs/android-g++/qmake.conf

```diff
@@ -63,7 +63,7 @@ contains(ANDROID_TARGET_ARCH, arm) {
     QMAKE_CFLAGS += -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16
     QMAKE_LFLAGS += -march=armv7-a -Wl,--fix-cortex-a8
   } else {
-    contains(ANDROID_TARGET_ABI, armeabi-v7a-hard) {
```

Comment:
I compare your patch with mine line by line, and
at now it looks strange:
contains(ANDROID_TARGET_ABI, armeabi-v7a) {
...
  } else {
    contains(ANDROID_TARGET_ABI, armeabi-v7a) {
  ... 
   } else {
...

why two times handle `contains(ANDROID_TARGET_ABI, armeabi-v7a) ` ?